docs/querying/nested-columns.md

```diff
@@ -0,0 +1,503 @@
+
+#### Example query: Extract nested data elements as suggested types
+
+The following example query illustrates how to use `JSON_VALUE` and the `RETURNING` keyword to extract an element of nested data and return it as specified types.
+
+> Note to self: add screenshot
+
+![Extract nested data elements as a suggested type](../assets/nested-extract-as-type.png)
+
+```sql
+SELECT
+  product,
+  department,
+  JSON_VALUE(shipTo, '$.address.country') as country,
+  JSON_VALUE(details, '$.price' RETURNING BIGINT) as price_int,
+  JSON_VALUE(details, '$.price' RETURNING DECIMAL) as price_decimal,
+  JSON_VALUE(details, '$.price' RETURNING VARCHAR) as price_varchar
+FROM nested_data_example
+```
+
+Query results:
+
+```json
+[["product","department","country","price_int","price_decimal","price_varchar"],["STRING","STRING","STRING","LONG","DOUBLE","STRING"],["VARCHAR","VARCHAR","VARCHAR","BIGINT","DECIMAL","VARCHAR"],["Bike","Sports","BL",542,542.0,"542.0"],["Bike","Sports","ME",955,955.0,"955.0"],["Sausages","Grocery","AD",8,8.0,"8.0"],["Mouse","Computers","CW",90,90.0,"90.0"],["Keyboard","Computers","TV",40,40.0,"40.0"]]
+```
+
+### Grouping, aggregating, and filtering
+
+You can use `JSON_VALUE` expressions in any context where you can use traditional Druid columns, such as grouping, aggregation, and filtering.
+
+#### Example query: Grouping and filtering
+
+The following example query illustrates how to use SUM, WHERE, GROUP BY and ORDER BY operators with `JSON_VALUE`.
+
+![Group, aggregate, filter](../assets/nested-group-aggregate.png)
+
+```sql
+SELECT
+  product,
+  JSON_VALUE(shipTo, '$.address.country'),
+  SUM(JSON_VALUE(details, '$.price' RETURNING BIGINT))
+FROM nested_data_example
+WHERE JSON_VALUE(shipTo, '$.address.country') in ('BL', 'CW')
+GROUP BY 1,2
+ORDER BY 3 DESC
+```
+
+Example query results:
+
+```json
+[["product","EXPR$1","EXPR$2"],["STRING","STRING","LONG"],["VARCHAR","VARCHAR","BIGINT"],["Bike","BL",542],["Mouse","CW",90]]
+```
+
+### Transforming JSON object data
+
+In addition to `JSON_VALUE`, Druid offers a number of operators that focus on transforming JSON object data: 
+
+- JSON_QUERY
+- JSON_OBJECT
+- PARSE_JSON
+- TO_JSON_STRING
+
+These functions are primarily intended for use with the Multi-Stage Query Architecture to transform data during insert operations, but they also work in traditional Druid SQL queries. Because most of these functions output JSON objects, they have the same limitations when used in traditional Druid queries as interacting with the JSON objects directly.
+
+#### Example query: Return results in a JSON object
+
+You can use the `JSON_QUERY` function to extract a partial structure from any JSON input and return results in a JSON object. Unlike `JSON_VALUE` it can extract objects and arrays.
+
+The following example query illustrates the differences in output between `JSON_VALUE` and `JSON_QUERY`.
```

Comment:
Add a line explaining the difference in output? Could be something like: The two output columns for `JSON_VALUE` contain null values only because `JSON_VALUE` only returns literal types.